Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -862,6 +862,10 @@ def line_problems(line, _lineno)
         end
       end
 
+      if line =~ /Utils\.(popen_(?:read|write))/
+        problem "Use `Utils.safe_#{Regexp.last_match(1)}` instead of `Utils.#{Regexp.last_match(1)}`"
```

Comment:
Would really like to see this added in RuboCop instead of another audit. I'm also not sure that I agree with this unconditionally; with audits like this we'd want to see all existing incidences fixed up in Homebrew/homebrew-core before they are added and merged here.

Reply:
I'll work on the RuboCop implementation. In terms of your concern for fixing in Homebrew/homebrew-core I would redirect you to the pipenv PR where these changes were suggested.

The impression I got there is that it was better to change the audit rules. This way, when the formulae are updated, the new audit will fail and prompt the change. I believe there are around 80 formulae that use `Utils.popen_read` so it would take some time to fix this for all of them (with a revision bump)

I'm sort of yielding because I don't know what's preferable and will go with whatever you and the maintainers say, so don't feel bad about telling me to wait or to do something different.

One of the other PRs changes `"SHELL=bash"` to `{"SHELL" => "bash"}`. There are only 4 formulae that require this change, so that would be easy to correct in homebrew-core (and I would be willing to fix those—it shouldn't take long).

Reply:
> I'll work on the RuboCop implementation.

Thanks ❤️ 



> The impression I got there is that it was better to change the audit rules.

Yeh, it's definitely better to add an audit. In general, though, we would rather try to fix all the cases at the same time as adding the rule and have it be a guide for new formulae when possible.

RuboCop can be configured to autocorrect these things too, which may make things easier.



> (and I would be willing to fix those—it shouldn't take long).

Great thanks ❤️

Reply:
> In general, though, we would rather try to fix all the cases at the same time as adding the rule and have it be a guide for new formulae when possible.

Makes sense. To help out, I can make a few PRs that make these changes on a few formulae as an example.

> RuboCop can be configured to autocorrect these things too, which may make things easier.

I'll look into this. That seems to be a good option that will make conversion easier (although it should be pretty straightforward to make these changes—they're pretty minor and have a message that tells you exactly what to do).

Reply:
> Makes sense. To help out, I can make a few PRs that make these changes on a few formulae as an example.

That would be great 🙇.



> I'll look into this. That seems to be a good option that will make conversion easier (although it should be pretty straightforward to make these changes—they're pretty minor and have a message that tells you exactly what to do).

Thanks again!